erdblick_app/app/sourcedata.panel.component.ts

```diff
@@ -0,0 +1,258 @@
+                                <p-treeTableToggler [rowNode]="rowNode" *ngIf="i == 0" />
+                                <span *ngIf="filterFields.indexOf(col.key) != -1" [innerHTML]="col.transform(rowData[col.key]) | highlight: filterString"></span>
+                                <span *ngIf="filterFields.indexOf(col.key) == -1" [innerHTML]="col.transform(rowData[col.key])"></span>
+                            </td>
+                        </tr>
+                    </ng-template>
+                </p-treeTable>
+            </ng-container>
+        </div>
+    
+        <ng-template #errorTemplate>
+            <div class="error">
+                <div>
+                    <strong>Error</strong><br>
+                    {{ errorMessage }}
+                </div>
+            </div>
+        </ng-template>
+    `
+})
+export class SourceDataPanelComponent implements OnInit {
+    @Input()
+    sourceData!: SelectedSourceData;
+
+    @ViewChild('tt')
+    table!: TreeTable;
+
+    treeData: TreeTableNode[] = [];
+    filterFields = [
+        "key",
+        "value"
+    ];
+    columns = [
+        { key: "key",     header: "Key",     width: '180px', transform: (v: any) => v },
+        { key: "value",   header: "Value",   width: '0*',    transform: (v: any) => v },
+        { key: "address", header: "Address", width: '80px',  transform: this.addressFormatter },
+        { key: "type",    header: "Type",    width: 'auto',  transform: this.schemaTypeURLFormatter },
+    ]
+
+    loading: boolean = true;
+    addressFormat: SourceDataAddressFormat = coreLib.SourceDataAddressFormat.BIT_RANGE;
+    errorMessage = "";
+    isExpanded = false;
+    filterString = "";
+
+    constructor(private inspectionService: InspectionService) {}
+
+    ngOnInit(): void {
+        this.inspectionService.loadSourceDataLayer(this.sourceData.tileId, this.sourceData.layerId, this.sourceData.mapId)
+            .then(layer => {
+                let root = layer.toObject()
+                this.addressFormat = layer.addressFormat();
+
+                layer.delete();
+
+                if (root) {
+                    this.treeData = root.children ? root.children : [root]
+                    this.selectItemWithAddress(this.sourceData.address);
+                } else {
+                    this.treeData = []
+                    this.setError('Empty layer.')
+                }
+            })
+            .catch(error => {
+                this.setError(`${error}`)
+            })
+            .finally(() => {
+                this.loading = false;
+            });
+    }
+
+    /**
+     * Set an error message that gets displayed.
+     * Unsets the tree to an empty array.
+     *
+     * @param message Error message
+     */
+    setError(message: string) {
+        this.loading = false;
+        this.treeData = [];
+        this.errorMessage = message;
+
+        console.error(this.errorMessage);
```

Comment:
Add a description. Something like `console.error("Error while processing SourceData tree:", this.errorMessage)`